src/Evaluation/EvaluationCalendar/WeeklyView/WeeklyView.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useMemo } from "react";
+import { Evaluation } from "../../EvaluationService";
+import { format } from "date-fns";
+
+interface WeeklyViewProps {
+  evaluations: Evaluation[];
+}
+
+const DATE_FORMAT = "EEE MMM dd yyyy";
+
+const getStartOfWeek = (date: Date): Date => {
+  const day = date.getDay(); // 0 (Sun) to 6 (Sat)
+  const diff = (day === 0 ? -6 : 1) - day; // Shift to Monday
+  const start = new Date(date);
+  start.setDate(start.getDate() + diff);
+  start.setHours(0, 0, 0, 0);
+  return start;
+};
+
+const addDays = (date: Date, days: number): Date => {
+  const result = new Date(date);
+  result.setDate(result.getDate() + days);
+  return result;
+};
+
+const WeeklyView: React.FC<WeeklyViewProps> = ({ evaluations }) => {
+  const startOfWeek = useMemo(() => {
+    if (evaluations.length === 0) {
+      return getStartOfWeek(new Date()); // fallback to current week
```

Comment:
this is an example of a good comment. but please be more elaborate.

```suggestion
      return getStartOfWeek(new Date()); // fallback to current week if no evaluation is found.
```

this might seem obvious right now, but as the code grows, this forms a direct connection between the if check and what you are returning.